zombies.js

```diff
@@ -78,7 +93,81 @@
  * @property {method} getPack              Returns private variable `pack`.
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
-
+class Player {
+  constructor(name, health, strength, speed){
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this._pack = [];
+    this._maxHealth = health;
+    this.isAlive = true;
+    this.equipped = false;
+  }
+  getPack(){
+    return this._pack;
+  }
+  getMaxHealth(){
+    return this._maxHealth;
+  }
+  takeItem(item){
+    if(this._pack.length === 3){
+      console.log("Pack is full, item could not be stored")
+      console.log(this.name + " has " + this._pack)
+      return false;
+    }else{
+      this._pack.push(item);
+      console.log(this.name + " has " + this._pack)
+      return true;
+    }
+  }
+  discardItem(item){
+    if(this._pack.indexOf(item) > -1){
+      this._pack.splice(this._pack.indexOf(item), 1);
+      console.log(this.name, this._pack)
+      return true;
+    }else{
+      console.log("You don't have that item");
+      return false;
+    }
+  }
+  checkPack(){
+    console.log(Player.getPack);
+  }
+  equip(itemToEquip){
+    if(itemToEquip instanceof Weapon && this._pack.includes(itemToEquip) && !this.equipped){
+      this.equipped = itemToEquip;
+      this._pack.splice(this._pack.indexOf(itemToEquip), 1);
+    }else if(itemToEquip instanceof Weapon && this._pack.includes(itemToEquip) && this.equipped){
+      this._pack.splice(this._pack.indexOf(itemToEquip), 1);
+      this._pack.push(this.equipped);
+      this.equipped = itemToEquip;
+    }
+  }
+  eat(itemToEat){
+    if(itemToEat instanceof Food && this._pack.includes(itemToEat) && (itemToEat.energy > (this._maxHealth - this.health) || this.health == this._maxHealth)){
```

Comment:
This could be simpler if you made the check for maxHealth after all the if statements.